k8s/upgrade/src/bin/upgrade-job/helm/values.rs

```diff
@@ -109,8 +110,21 @@ pub(crate) fn generate_values_yaml_file(
         upgrade_values_file.path(),
     )?;
 
+    // override the values with data passed as --set flag during upgrade
+
     // helm upgrade .. --set image.tag=<version> --set image.repoTags.controlPlane= --set
     // image.repoTags.dataPlane= --set image.repoTags.extensions=
 
+    // Override all the values by the value passed in set flag
+    let parts: Vec<&str> = values.split(',').collect();
+    for tokens in parts {
+        let data: Vec<&str> = tokens.split('=').collect();
+        yq.set_string_value(
+            YamlKey::try_from(format!(".{}", data[0]).as_str())?,
+            data[1],
+            upgrade_values_file.path(),
+        )?;
```

Comment:
This will panic if index 1 or 0 don't exist.
You can get around that using slice pattern matching, example:
```rust
      let v = l.split("=").collect::<Vec<_>>(); 
      match v[..] {
        [ var, value ] => println!("{var}={value}"),
        _ => println!("Invalid!")
      }
```
But we have to consider here what kind of inputs we allow. What if we have ',' in the values, example: --set="v={1,2}" ?

Reply:
@tiagolobocastro  as discussed the helm upgrade dry-run command validates the values. 
PTAL